src/dataset-builder/ConceptSearch.test.ts

```diff
@@ -0,0 +1,187 @@
+    expect(mockSearch).toHaveBeenCalledWith(domainOption.root, searchText);
+  });
+
+  it('filters based on search text', async () => {
+    // Arrange/Act
+    const searchText = 'search text';
+    // Need to explicitly call act() here to avoid a warning about state updates during rendering.
+    await act(() => renderSearch(searchText));
+    // Assert - the initial page load should have called search with the initial search text.
+    expect(mockSearch).toHaveBeenCalledWith(domainOption.root, searchText);
+  });
+
+  it('calls open hierarchy on click', async () => {
+    // Arrange
+    const concept = displayedConcepts[0];
+    renderSearch();
+    await screen.findByText(concept.name);
+    // Act
+    const user = userEvent.setup();
+    await user.click(screen.getByLabelText(`add ${concept.id}`));
+    await user.click(screen.getByLabelText(`open hierarchy ${concept.id}`));
+    // Assert
+    expect(onOpenHierarchy).toHaveBeenCalledWith(
+      { id: concept.id, category: domainOption.category, root: concept },
+      [concept],
+      ''
+    );
+  });
+
+  it('supports add to cart', async () => {
+    // Arrange
+    renderSearch();
+    const concept = displayedConcepts[0];
+    await screen.findByText(concept.name);
+    // Act
+    const user = userEvent.setup();
+    await user.click(screen.getByLabelText(`add ${concept.id}`));
+    // Assert
+    expect(screen.queryByText(actionText)).toBeTruthy();
+    expect(screen.queryByText('1 concept', { exact: false })).toBeTruthy();
+  });
+
+  it('supports remove from cart', async () => {
+    // Arrange
+    renderSearch();
+    const concept = displayedConcepts[0];
+    await screen.findByText(concept.name);
+    // Act
+    const user = userEvent.setup();
+    await user.click(screen.getByLabelText(`add ${concept.id}`));
+    await user.click(screen.getByLabelText(`remove ${concept.id}`));
+    // Assert
+    expect(screen.queryByText(actionText)).toBeFalsy();
+    expect(screen.queryByText('1 concept', { exact: false })).toBeFalsy();
+  });
+
+  it('calls commit on action', async () => {
+    // Arrange
+    renderSearch();
+    const concept = displayedConcepts[0];
+    await screen.findByText(concept.name);
+    // Act
+    const user = userEvent.setup();
+    await user.click(screen.getByLabelText(`add ${concept.id}`));
+    await user.click(screen.getByText(actionText));
+    // Assert
+    expect(onCommit).toHaveBeenCalledWith([concept]);
+  });
+
+  it('testing HighlightConceptName, "Dis"', async () => {
+    renderSearch('Dis');
+
+    const disText = await screen.findAllByText('Dis');
+
+    const filterDisText = _.filter(
+      (element) => element.tagName === 'DIV' && element.style.fontWeight === '600',
+      disText
+    ).length;
+    expect(filterDisText).toBeGreaterThan(0);
+    expect(await screen.findByText('ease')).toBeTruthy();
+  });
+
+  it('testing HighlightConceptName, Checking that spaces still exist on both sides', async () => {
```

Comment:
Same as above, how would you feel about leaving out the component name and just including the `'spaces still exist on both sides when searching'`